### PR TITLE
Fixing Public Jira Issue : APIMANAGER-3945

### PR DESCRIPTION
**Pull request by ChamNDeSilva**
Get subscriptions by providing API

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag [hunk 1]
```diff
@@ -144,7 +144,58 @@ include("/jagg/jagg.jag");
...
+        		print({
+        		    error:true,
+        		    message:msg.error.loginRequired(action)
+                    });
+        	}
+        	return;
+        }
+        username = user.username;
+        var log = new Log();
+        log.info(username);
```
Comment (harsha89) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag]: may be this should be a debug log or we can remove this if this is for testing purposes. 
It's best to initialize the log with the name to identify where it cause the issue like var log = new Log('subscriptions list');

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag [hunk 2]
```diff
@@ -144,7 +144,56 @@ include("/jagg/jagg.jag");
...
+        		print({
+        		    error:true,
+        		    message:msg.error.loginRequired(action)
+                    });
+        	}
+        	return;
+        }
+        username = user.username;
+
+        if((request.getParameter('apiName')!=null) && (request.getParameter('apiVersion')!=null) &&
```
Comment (lalaji) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag]: assigning request parameters to variables [eg:apiName = request.getParameter('apiName')] can be done before the if statement.Then those variables can be directly use from if statement and inside 'if' condition.

Comment (ChamNDeSilva) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag]: Fixed in PR https://github.com/wso2/carbon-apimgt/pull/1093

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag [hunk 3]
```diff
@@ -104,7 +104,7 @@ include("/jagg/jagg.jag");
             }
         }
         print(obj);
-     }else if (action == "getSubscriptionByApplication") {
+    }else if (action == "getSubscriptionByApplication") {
```
Comment (prabathabey) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag]: Formatting issues.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag [hunk 4]
```diff
@@ -144,7 +144,56 @@ include("/jagg/jagg.jag");
             }
         }
         print(obj);
-     }else {
+    }else if (action == "getSubscriptionByAPI"){
```
Comment (prabathabey) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag]: More formatting issues.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag [hunk 5]
```diff
@@ -144,7 +144,56 @@ include("/jagg/jagg.jag");
...
+            }
+        }else{
+            obj = {
+                error:result.error,
+                message:msg.error.invalidInput(action, username)
+            };
+        }
+        print(obj);
+
+    }else {
```
Comment (prabathabey) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag]: Some more formatting issues.

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag [hunk 6]
```diff
@@ -144,7 +144,56 @@ include("/jagg/jagg.jag");
...
+                    error:result.error,
+                    message:msg.error.authError(action, username)
+                };
+            } else {
+                obj = {
+                    error:false,
+                    applications:result.subscriptions
+                }
+            }
+        }else{
```
Comment (prabathabey) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/subscription/subscription-list/ajax/subscription-list.jag]: Formatting issues.

**Pull request merged**